common/src/main/java/org/terracotta/angela/common/distribution/WatchedProcess.java

```diff
@@ -30,24 +31,24 @@
   private final int pid;
 
   public WatchedProcess(ProcessExecutor processExecutor, final AtomicReference<S> stateRef, final S deadState) {
+    processExecutor.addListener(new ProcessListener() {
+      @Override
+      public void afterFinish(Process process, ProcessResult result) {
+        stateRef.set(deadState);
+      }
+
+      @Override
+      public void afterStop(Process process) {
+        stateRef.set(deadState);
+      }
+    });
+
     try {
       this.startedProcess = processExecutor.start();
     } catch (IOException e) {
       throw new RuntimeException("Cannot start process " + processExecutor.getCommand(), e);
     }
     this.pid = PidUtil.getPid(startedProcess.getProcess());
-
-    Thread watcherThread = new Thread(() -> {
```

Comment:
No need for an additional watcher daemon thread: ZT lib already has a listener mechanism and already spawns a thread,